backend/models/refund_request.py

```
@@ -0,0 +1,756 @@
+                }
+            }
+
+            updated_tx = await transactions_col.find_one_and_update(
+                atomic_filter,
+                {"$inc": {"refunded_amount": final_refund_amount}},
+                return_document=ReturnDocument.AFTER
+            )
+            if not updated_tx:
+                logging.error(f"Atomic reservation failed: refund of ${final_refund_amount} would exceed transaction {transaction.get('transaction_id')} amount ${total_transaction_amount}")
+                return None
+        except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)